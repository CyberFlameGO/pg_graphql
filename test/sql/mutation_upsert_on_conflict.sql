begin;

    create table account(
        id serial primary key,
        email varchar(255) not null unique,
        to_change int not null
    );

    insert into public.account(email, to_change)
    values
        ('aardvark@x.com', 1);


    select graphql.resolve($$
    mutation {
      upsertAccount(
        object: {
            email: "aardvark@x.com"
            toChange: 2
        },
        onConflict: {
            conflictFields: ["id"]
            updateFields: ["toChange"]
        }
      ) {
        email
        toChange
      }
    }
    $$);

rollback;
