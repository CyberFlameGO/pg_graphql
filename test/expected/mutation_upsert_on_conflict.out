begin;
    create table account(
        id serial primary key,
        email varchar(255) not null unique
        to_change int not null
    );
ERROR:  syntax error at or near "to_change"
LINE 4:         to_change int not null
                ^
    insert into public.account(email, to_change)
    values
        ('aardvark@x.com', 1);
ERROR:  current transaction is aborted, commands ignored until end of transaction block
    
    select graphql.resolve($$
    mutation {
      upsertAccount(
        object: {
            email: "aardvark@x.com"
            toChange: 2
        },
        onConflict: {
            conflictFields: ["id"]
            updateFields: ["toChange"]
        }
      }) {
        email
        toChange
      }
    }
    $$);
ERROR:  current transaction is aborted, commands ignored until end of transaction block
rollback;
